File: apps/mobile/src/lib/nutrition/logic/candidate-nutrients.ts
/**
 * The supported food-source-candidate nutrients, vendored as consts from web
 * `lib/nutrition/catalog/nutrients.ts` (keep in sync). Only the candidate
 * gating is needed on mobile — `NUTRIENT_META`/`DEFAULT_NUTRIENTS` are not.
 */
import type { SupportedCandidateNutrient } from '@/lib/api/contracts/nutrition';
import type { NutritionNutrientKey } from '@/lib/nutrition/types';

export const SUPPORTED_CANDIDATE_NUTRIENTS = [
  'calciumMg',
  'ironMg',
  'vitaminCMg',
  'phosphorusMg',
  'vitaminB1Mg',
  'vitaminB2Mg',
  'vitaminPpMg',
  'vitaminAMcg',
] as const satisfies readonly SupportedCandidateNutrient[];

export const SUPPORTED_CANDIDATE_NUTRIENT_SET: ReadonlySet<string> = new Set(
  SUPPORTED_CANDIDATE_NUTRIENTS
);

/** Narrows a nutrient key to a supported candidate nutrient, or null. */
export function asSupported(
  nutrient: NutritionNutrientKey
): SupportedCandidateNutrient | null {
  return SUPPORTED_CANDIDATE_NUTRIENT_SET.has(nutrient)
    ? (nutrient as SupportedCandidateNutrient)
    : null;
}

File: apps/mobile-flutter/lib/features/nutrition/logic/candidate_nutrients.dart
/// The supported food-source-candidate nutrients, vendored as consts from web
/// `lib/nutrition/catalog/nutrients.ts` (keep in sync). Only the candidate
/// gating is needed on mobile.
///
/// Ported from `apps/mobile/src/lib/nutrition/logic/candidate-nutrients.ts`.
library;

import '../../../models/nutrition.dart';

/// The supported candidate nutrient enum names, as raw strings (matching the
/// REST contract's `SupportedCandidateNutrient`).
const List<String> kSupportedCandidateNutrients = [
  'calciumMg',
  'ironMg',
  'vitaminCMg',
  'phosphorusMg',
  'vitaminB1Mg',
  'vitaminB2Mg',
  'vitaminPpMg',
  'vitaminAMcg',
];

final Set<String> _supportedSet = kSupportedCandidateNutrients.toSet();

/// Narrows a nutrient key to a supported candidate nutrient string, or null.
String? asSupported(NutritionNutrientKey nutrient) {
  final name = nutrient.name;
  return _supportedSet.contains(name) ? name : null;
}
